File: src/binary_search/binary_search.go
package binarysearch

func binarySearch(haystack []int, needle int) bool {
	low := 0
	high := len(haystack)

	for low < high {
		mid := (high + low) / 2
		if haystack[mid] == needle {
			return true
		} else if haystack[mid] > needle {
			high = mid
		} else {
			low = mid + 1
		}
	}
	return false
}

File: src/binary_search/binary_search.dart
bool binarySearch(int needle, List<int> haystack) {
  int high = haystack.length;
  int low = 0;

  while (low < high) {
    int mid = (low + high) ~/ 2;

    if (haystack[mid] == needle) {
      return true;
    } else if (haystack[mid] > needle) {
      high = mid;
    } else if (haystack[mid] < needle) {
      low = mid + 1;
    }
  }

  return false;
}
